### PR TITLE
mgr/dashboard: Display logged in user

### PR DESCRIPTION
**Pull request by familyuu**
mgr/dashboard: Display logged in user

Fixes: http://tracker.ceph.com/issues/24822

Signed-off-by: Dan Guo <familyuu@163.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 1]
```diff
@@ -224,7 +224,7 @@
     .datatable-pager .pager {
       margin-right: 5px !important;
       & li:not(:first-child) {
-        /** Add space between buttons */
+        /* * Add space between buttons */
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss]: Is this change intentional?

**Comment by LenzGr** [comment]
Thanks again for submitting this PR and for making the requested changes, much appreciated. Please don't close this PR if further changes are needed, but force push your updated changes into the branch on your personal git repository. Hopefully we're not too far away from getting this feature merged - at least "make check" has run successfully now :)

File: src/pybind/mgr/dashboard/frontend/package-lock.json [hunk 2]
```diff
@@ -4170,12 +4170,14 @@
         "balanced-match": {
           "version": "1.0.0",
           "bundled": true,
-          "dev": true
+          "dev": true,
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/package-lock.json]: Please discard all the changes in this file.
For some reason, sometimes npm adds this changes. This can be removed by running 'npm install' again.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 3]
```diff
@@ -249,7 +249,6 @@
     }
   }
 }
-
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss]: Please remove this change.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss (hunk 1 above)
Comment (familyuu) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss]: No， it changed automatically after running the command 'npm run prettier' in my development platform.

**Comment by LenzGr** [comment]
> Since this will replace the old logout component, you should remove that component.

I agree - @familyuu can you please take care of this in your PR?

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/identity/identity.component.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<div dropdown>
+  <a dropdownToggle class="dropdown-toggle" data-toggle="dropdown" i18n-title title="Logged in user">
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/identity/identity.component.html]: List every tag attribute in a separate line.
Please do this in all HTML files.

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/identity/identity.component.html [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    <li>
+      <a i18n class="dropdown-item" href="#">Signed in as
+        <strong>{{ username }}</strong>
+      </a>
+    </li>
+    <li class="divider dropdown-divider"></li>
+    <li>
+      <a i18n class="dropdown-item" (click)="logout()">
+        <i class="fa fa-sign-out fa-fw"></i>
+        <span> Sign out</span>
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/identity/identity.component.html]: Missing ``ì18n`` attribute.

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/identity/identity.component.html [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  </a>
+  <ul *dropdownMenu class="dropdown-menu dropdown-menu-right" role="menu">
+    <li>
+      <a i18n class="dropdown-item" href="#">Signed in as
+        <strong>{{ username }}</strong>
+      </a>
+    </li>
+    <li class="divider dropdown-divider"></li>
+    <li>
+      <a i18n class="dropdown-item" (click)="logout()">
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/identity/identity.component.html]: Relocate the i18n attribute to the <span> tag.

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/identity/identity.component.html [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<div dropdown>
+  <a dropdownToggle class="dropdown-toggle" data-toggle="dropdown" i18n-title title="Logged in user">
+    <i class="fa fa-user fa-fw"></i>
+    <span class="caret"></span>
+  </a>
+  <ul *dropdownMenu class="dropdown-menu dropdown-menu-right" role="menu">
+    <li>
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/identity/identity.component.html]: I suggest to use ``<li class="disabled">`` to disable the selection of the menu item. See https://getbootstrap.com/docs/3.3/components/#dropdowns-disabled.

**Comment by LenzGr** [comment]
I tested your pull request on my local dev environment and it looks good. Thanks a lot!

I would only have one small enhancement request: currently, moving the mouse over the "Signed in as <username>" line makes it highlighted, as if there is any link behind it. However, clicking on the line does not result in any action. I would therefore recommend to disable the highlighting of that line.

screenshot from 2018-09-25 16-07-27

**Comment by callithea** [comment]
jenkins test dashboard

**Comment by callithea** [comment]
jenkins test dashboard

**Comment by LenzGr** [comment]
Ping @familyuu - can you please look into the remaining requested changes? I think we're close to getting this PR merged :)

**Comment by familyuu** [comment]
I have modified code with the recommendations of all reviewers, and git push to my branch. 
Since the Jenkins ceph dashboard test failed, I checked the error log, it is something about network was wrong. Anyone has some good idea?

**Comment by callithea** [comment]
> I have modified code with the recommendations of all reviewers, and git push to my branch.
> Since the Jenkins ceph dashboard test failed, I checked the error log, it is something about network was wrong. Anyone has some good idea?

@familyuu no worries, I was just testing a new check for our dashboard frontend (e2e) tests - it's not a required check, so your pull request can be merged even if the dashboard check failed. Sorry for the noise :)

**Comment by LenzGr** [comment]
Thanks for the update @familyuu - looks good from my end user's perspective. I think you still need to address the [request](https://github.com/ceph/ceph/pull/24213#pullrequestreview-158033781) from @tspmelo to actually remove the old logout component.

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/identity/identity.component.html [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+         href="#">Signed in as
+        <strong>{{ username }}</strong>
+      </a>
+    </li>
+    <li class="divider dropdown-divider"></li>
+    <li role="menuitem">
+      <a class="dropdown-item"
+         (click)="logout()">
+        <i class="fa fa-sign-out fa-fw"></i>
+        <span i18n> Sign out</span>
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/identity/identity.component.html]: Is the white space before 'Sign' necessary?

Comment (ricardoasmarques) [reply src/pybind/mgr/dashboard/frontend/src/app/core/navigation/identity/identity.component.html]: @familyuu can you please remove this unnecessary white space? After that we can merge this PR.
